posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc

```diff
@@ -0,0 +1,145 @@
+With Open Liberty 20.0.0.9 you can now experiment with the type-safe SmallRye GraphQL Client API, and more easily write and run GraphQL queries and mutations using a built-in GraphiQL user interface. Plus there has been a number of significant bug fixes.
+
+In link:{url-about}[Open Liberty] 20.0.0.9:
+
+* <<GraphQLAPIs, Experiment with a third-party GraphQL client API>>
+* <<GraphiQL, Built-in GraphiQL user interface for faster queries and mutations>>
+* <<bugs, Significant bugs fixed in this release>>
+
+// end::intro[]
+// tag::run[]
+[#run]
+
+== Run your apps using 20.0.0.9
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+[source,xml]
+----
+<dependency>
+    <groupId>io.openliberty</groupId>
+    <artifactId>openliberty-runtime</artifactId>
+    <version>20.0.0.9</version>
+    <type>zip</type>
+</dependency>
+----
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[20.0.0.9,)'
+}
+----
+Or if you're using Docker:
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+[#GraphQLAPIs]
+== Experiment with a third-party GraphQL client API
+
+MicroProfile GraphQL has only been available in Open Liberty for a few months now, and it is already a hit! That said, there are lots of things we'd like to do to improve it - and make it more complete. One of those things is a Client API. While the official Client API is not expected until the next release of MicroProfile GraphQL, you can now experiment with the type-safe SmallRye GraphQL Client API. SmallRye is the underlying implementation of MicroProfile GraphQL, and you can access some of its above-and-beyond-the-spec features by adding the "third-party" API type visibility to your application, like so:
+[source,xml]
+----
+<application name="MyGraphQLApp" location="MyGraphQLApp.war">
+    <classloader apiTypeVisibility="+third-party"/>
+</application>
+----
+Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql
+The SmallRye GraphQL Client APIs are very similar to https://github.com/eclipse/microprofile-rest-client[MicroProfile Rest Client] in that it uses an interface to invoke remote methods in a type-safe manner. For example, suppose we want a client that can invoke a query of all super heroes in a given location.  We would create a query interface like this:
+[source,java]
+----
+@GraphQlClientApi
+interface SuperHeroesApi {
+    List<SuperHero> allHeroesIn(String location); // (1)
```

Comment:
```suggestion
    List<SuperHero> allHeroesIn(String location);
```

removing the unnecessary comment - sorry, I had left that in there from the example code in my original post.

Reply:
Done